readthedocs/core/logs.py

```diff
@@ -69,8 +68,14 @@ def __call__(self, logger, method_name, event_dict):
             "line.number": record.lineno,
         }
 
+        # NOTE: For some reason structlog isn't including
+        # the exception in the record.exc_info attribute,
+        # so we need to get it from the event_dict.
+        exc_info = event_dict.pop("exc_info", None)
         if record.exc_info:
             output.update(format_exc_info(record.exc_info))
+        elif exc_info:
+            output.update(format_exc_info(_figure_out_exc_info(exc_info)))
```

Comment:
It's like you can strip out the essential part of the internal structlog function `_figure_out_exc_info` and it makes things less confusing...

```suggestion
        elif isinstance(exc_info, BaseException):
            output.update(
                format_exc_info(
                    (exc_info.__class__, exc_info, getattr(exc_info, "__traceback__"))
                )
            )
        elif isinstance(exc_info, tuple):
            output.update(format_exc_info(exc_info))         
```

Reply:
Maybe this part should also be there: https://github.com/hynek/structlog/blob/bcfc7f9e60640c150bffbdaeed6328e582f93d1e/src/structlog/processors.py#L528-L529